### ISSUE TITLE
Auto update torrent title

### ISSUE THREAD

**Issue by impalex** [werwolfby/monitorrent#173]
Предложение: сделать возможность автоматического обновления названия торрента. Разумеется, опционально.

Релизеры, как правило, обновляют количество серий в заголовке, тут же все остается замороженным на момент добавления топика. Поэтому не видим реального состояния раздачи, стоит ли дальше следить или нет.

**Comment by werwolfby** [comment]
Согласен у нас это даже как-то реализованно. Но сейчас проблема в том, что мы можем руками указывать название топика. И как быть при автоматической обновлении?

Если делать это опционально, то не давать пользователю его редактировать. 
Если обновился только заголовок, но не обновился сам торрент, то не обновлять дату обновления топика.

Наверное так можно будет сделать, вполне.

Не для всех трекеров это будет и не для всех должно работать. Например для лостфильма и ньюстудио это всегда имя сериала и смысла в обновлении заголовка нет. 

С другой стороны у нас есть поле info которое возвращается на фронт энд и пока никак не отображается. Идея как раз и была в том чтобы в этом поле указывать дополнительную информацию: https://github.com/werwolfby/monitorrent/blob/b31dd9037d789f08184a382c64afa29dd81a4091/monitorrent/plugins/trackers/__init__.py#L123

Реализовано только в lostfilm плагине и [выдает](https://github.com/werwolfby/monitorrent/blob/23837c21e4346c70c325ed4f16a763e17f75046c/monitorrent/plugins/trackers/lostfilm.py#L577-L582): SxxExx, т.е. последний скачанный эпизод. Но как я говорил на UI мы пока это не отображаем. А на других трекерах выдрать эту инфу гораздо сложнее. С этим и борюсь в разработке ньюстудио плагина.

Но думаю ваше предложение более эффективно. Стоит подумать.

**Comment by impalex** [comment]
Да, я обратил внимание на доп. поле в плагине лостфильма, когда изучал плагины. Для "обычных" трекеров такой подход не пойдёт.

Парсить названия тем - занятие сильно неблагодарное. Формат указания списка серий может различаться от темы к теме. А ведь мы можем добавить в отслеживание и не только сериалы - ну мало ли, софт, книги, музыку и т.д. и т.п.

Я вижу два возможных варианта:

1) Чекбокс в настройках торрента. Как сделано в TorrentMonitor (http://img.impa.me/k7gQW). Всё прямо и просто.
2) А можно поступить чуть хитрее... никаких чекбоксов. Если пользователь поменял распарсеное плагином наименование - ему виднее, используем вариант пользователя. Если пользователь не стал трогать то, что напарсил плагин - отслеживаем и изменения названия. Но логика получается не такая прозрачная как в первом варианте...

Вариант с чекбоксом как-то более симпатичен. Во втором слишком много нюансов.

**Comment by werwolfby** [comment]
Согласен. И парсин топика не благодарен - это тоже понятно, но ему все равно быть :)
Будем пробовать выщемлять как можно больше инфы из заголовка, а потом и из страници. 

Сделаем чекбок для всех ```WithHasChanged``` плагинов

**Comment by werwolfby** [comment]
Ишью отправляется в разработку, и так:

- [ ] Добавляем чекбокс как в Torrent Monitor: **Update title automatically**
- [ ] Апдейт заголовка может происходить не зависимо от самого торрент файла, и не влияет на Last Update
- [ ] По умолчанию у всех уже добавленых торрентов **Update title automatically** отключен

@DSilence @kijojo @impalex @Schoshka @abaistov Есть долнения? Особенно по последнему пункту? Может как раз таки включать автоматически?

**Comment by impalex** [comment]
У меня дополнений нет. По последнему пункту - всё верно, те, что уже были созданы, лучше оставить без изменений. Всё-таки на данный момент дефолтным поведением является сохранение названия без изменений, не стоит его менять.

**Comment by abaistov** [comment]
Данную опцию буду включать на всех раздачах, поэтому для меня было бы удобней , чтобы для добавленных топиков данная опция была включена. А вообще, это не так уж и принципиально , "галочки" раставить - минутное дело :)

**Comment by Shooshka** [comment]
Ну если "Для добавляемых по умолчанию включена", то почему тогда "По умолчанию у всех уже добавленых торрентов Update title automatically отключен"? :)

Да и как-то странно, даже если человеку не нужна эта фишка и пофиг ему, что в титле, например, "1 серия из 20", когда сезон давно закончен, как она может помешать? Голос за включение по умолчанию и включение у всех уже добавленных в общем.

**Comment by werwolfby** [comment]
Потому что гланый принцип "не навреди": 
* когда при добавлении она включены, вы это видете и можете убрать, но по умолчанию она должна включаться
* когда же я (а я так и делаю), руками меняю тайт и убираю инфу о том сколько серий из сезона сейчас в раздаче, то после апдейта на новую версию, я не хочу чтобы это автоматически поменялось, а автоматически мы не можем сейчас определить руками менялся тайтл или нет.

Поэтому я хочу оставить эту вичу включаемой для старых раздач, а для новых пользователь увидит, что галочка установлена.

**Comment by werwolfby** [comment]
Но я прекрасно осознаю что большинство ничего не меняет поэтому им хотелось бы проапдейтать тайтлы автоматически.

Я вообще наверное единственный который руками правет тайтл топика 😄

**Comment by werwolfby** [comment]
Ну и последний минус, на самой странице торрента, не узнать одновился он или нет (я это проверю, как минимум не на всех трекерах).
А для проверки сейчас качается сам торрент файл, в большинстве случаев это банально быстрее, потому что торрент файл весит гораздо меньше HTML странички раздачи.
И уже по этому файлу определялось, поменялся торрент или нет.
Сейчас же придётся делать 2 запроса, первый за изменением заголовка, второй за торрент файлом, или апдейтать заголовоко только если торрент файл поменялся (и только тогда делать второй запрос).

**Comment by werwolfby** [comment]
Хотя вот на рутрекере и по магнет ссылки и по зарегистрирован вроде как можно понять апдейтался или нет торрент.

**Comment by impalex** [comment]
> Сейчас же придётся делать 2 запроса, первый за изменением заголовка, второй за торрент файлом, или апдейтать заголовоко только если торрент файл поменялся (и только тогда делать второй запрос).

Издержки.
Для анидаба, к примеру, и так приходится делать два запроса - первый за страницей, оттуда выдрать ссылку на торрент, потом уж сам торрент. Просто там ссылка на торрент не статична и меняется при обновлении. Так и чешется где-то между этими двумя запросами обновить и заголовок :)

**Comment by werwolfby** [comment]
Для кинозала тоже будет работать, там вообще это вынужденая мера так как есть ограничение на количество скачанных торрентов в день.

**Comment by Egorkat** [comment]
Спасибо за проект! Пользуемся с удовольствием и ждем реализации этого функционала
